### PR TITLE
Shoatman add client info to refresh token request

### PR DESCRIPTION
**Pull request by shoatman**
Added client_info to token response.  
Corrected unit tests broken by this change and by the change in common for null user cache entry.

### CODE REVIEW

File: adal/src/androidTest/java/com/microsoft/aad/adal/Util.java [hunk 1]
```diff
@@ -159,10 +159,11 @@ static String getErrorResponseBody(final String errorCode) throws JSONException
 
     static byte[] getPoseMessage(final String refreshToken, final String clientId, final String resource)
             throws UnsupportedEncodingException {
-        return String.format("%s=%s&%s=%s&%s=%s&%s=%s",
+        return String.format("%s=%s&%s=%s&%s=%s&%s=%s&%s=%s",
                 AuthenticationConstants.OAuth2.GRANT_TYPE, urlFormEncode(AuthenticationConstants.OAuth2.REFRESH_TOKEN),
                 AuthenticationConstants.OAuth2.REFRESH_TOKEN, urlFormEncode(refreshToken),
                 AuthenticationConstants.OAuth2.CLIENT_ID, urlFormEncode(clientId),
+                AuthenticationConstants.OAuth2.CLIENT_INFO, urlFormEncode(AuthenticationConstants.OAuth2.CLIENT_INFO_TRUE),
```
Comment (iambmelt) [inline adal/src/androidTest/java/com/microsoft/aad/adal/Util.java]: Sorry to have missed this. Thanks @shoatman

File: adal/src/main/java/com/microsoft/aad/adal/Oauth2.java [hunk 2]
```diff
@@ -222,15 +222,20 @@ public String buildTokenRequestMessage(String code) throws UnsupportedEncodingEx
...
                 AuthenticationConstants.OAuth2.GRANT_TYPE,
                 StringExtensions.urlFormEncode(AuthenticationConstants.OAuth2.REFRESH_TOKEN),
 
                 AuthenticationConstants.OAuth2.REFRESH_TOKEN,
                 StringExtensions.urlFormEncode(refreshToken),
 
                 AuthenticationConstants.OAuth2.CLIENT_ID,
-                StringExtensions.urlFormEncode(mRequest.getClientId()));
+                StringExtensions.urlFormEncode(mRequest.getClientId()
+                ),
```
Comment (iambmelt) [inline adal/src/main/java/com/microsoft/aad/adal/Oauth2.java]: Formatting here is a little goofy

File: adal/src/main/java/com/microsoft/aad/adal/TokenCacheAccessor.java [hunk 3]
```diff
@@ -300,7 +300,7 @@ void updateTokenCache(final String resource, final String clientId, final Authen
             return;
         }
 
-        if (mUseCommonCache == true) {
+        if (mUseCommonCache && !UrlExtensions.isADFSAuthority(new URL(mAuthority))) {
```
Comment (iambmelt) [inline adal/src/main/java/com/microsoft/aad/adal/TokenCacheAccessor.java]: 👍

**Comment by iambmelt** [comment]
This PR looks to also be addressing https://github.com/AzureAD/azure-activedirectory-library-for-android/issues/1182

File: build.gradle [hunk 4]
```diff
@@ -7,7 +7,7 @@ buildscript {
     }
 
     dependencies {
-        classpath "com.android.tools.build:gradle:$rootProject.ext.gradleVersion"
+        classpath 'com.android.tools.build:gradle:3.1.2'
```
Comment (shoatman) [inline build.gradle]: Need to fix this.  This is Android Studio's attempt to fix for me.

**Pull request closed without merging**